python/pyarrow/_exec_plan.pyx

```diff
@@ -95,6 +98,23 @@ cdef execplan(inputs, output_type, vector[CDeclaration] plan, c_bool use_threads
                 c_in_table, 1 << 20)
             c_input_node_opts = static_pointer_cast[CExecNodeOptions, CTableSourceNodeOptions](
                 c_tablesourceopts)
+        elif isinstance(ipt, FilteredDataset):
```

Comment:
I wonder why the existing code for Dataset below wouldn't work here. Especially, one creates a Source node, the other a Scan node. Is there a functional or performance difference? cc @westonpace for insight.

Reply:
While there shouldn't be a significant performance difference this does seem a little less than ideal.  Could the `_make_scanner` method be changed to a method that creates scan options instead?

Also, if the user only asks for some columns does the projection get passed down into the `ToRecordBatchReader` call?

Reply:
There shouldn't be any API on Dataset at the moment that should allow you to restrict the columns before executing a join through the exec plan. We don't yet provide any "projection" capability for `FilteredDataset` itself, only for scanner.

Making ScanOptions seems like a good idea btw, would make easy to expand the capabilities in the future. Adding projections for example would be just one more passed option.

Reply:
The code has been updated to now use ScanOptions and rely on those for the compute engine plan.